### PR TITLE
Add libkmod

### PR DESCRIPTION
**Pull request by zhangxp1998**
Add libkmod , a library for reading/manipulating kernel modules.

### CODE REVIEW

File: packages/l/libkmod/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    add_versions("v30", "1fa3974abd80b992d61324bcc04fa65ea96cfe2e9e1150f48394833030c4b583")
+
+    on_install("linux", "android", function (package)
+		os.cp(path.join(package:scriptdir(), "port", "xmake.lua"), "xmake.lua")
+		os.cp(path.join(package:scriptdir(), "port", "config.h"), "config.h")
+		os.cp(path.join(package:scriptdir(), "port", "elf.h"), "elf.h")
+		os.cp(path.join(package:scriptdir(), "port", "endian-darwin.h"), "endian-darwin.h")
+
+        local configs = {}
+		if package:config("shared") then
```
Comment (waruqi) [inline packages/l/libkmod/xmake.lua:17]: incorrect indent, use 4 spaces

File: packages/l/libkmod/port/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+
+target("kmod")
+	set_kind("$(kind)")
```
Comment (waruqi) [inline packages/l/libkmod/port/xmake.lua:4]: and here, indent

File: packages/l/libkmod/port/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (waruqi) [inline packages/l/libkmod/port/xmake.lua:1]: please add `add_rules("mode.debug", "mode.release")`

File: packages/l/libkmod/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    set_license("LGPL-2.1")
+
+    add_urls("https://github.com/kmod-project/kmod/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/kmod-project/kmod.git")
+    add_versions("v30", "1fa3974abd80b992d61324bcc04fa65ea96cfe2e9e1150f48394833030c4b583")
+
+    on_install("linux", "android", function (package)
+		os.cp(path.join(package:scriptdir(), "port", "xmake.lua"), "xmake.lua")
+		os.cp(path.join(package:scriptdir(), "port", "config.h"), "config.h")
+		os.cp(path.join(package:scriptdir(), "port", "elf.h"), "elf.h")
```
Comment (waruqi) [inline packages/l/libkmod/xmake.lua:13]: why not use `add_deps("libelf")` or `add_deps("elfutils")`?

File: packages/l/libkmod/port/elf.h [hunk 5]
```diff
@@ -0,0 +1,3558 @@
+/* This file defines standard ELF types, structures, and macros.
+   Copyright (C) 1995-2015 Free Software Foundation, Inc.
```
Comment (waruqi) [inline packages/l/libkmod/port/elf.h:2]: it's too large. please use `libelf` or `elfutils` package

File: packages/l/libkmod/port/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+target("kmod")
+	set_kind("$(kind)")
+	set_languages("gnu99")
+	add_headerfiles("(libkmod/libkmod.h)")
+	add_headerfiles("(libkmod/libkmod-index.h)")
+	add_includedirs(".")
+	add_defines("PATH_MAX=4096")
+	add_defines("ANOTHER_BRICK_IN_THE")
+	add_defines("SYSCONFDIR=\"/tmp\"")
+	add_defines("HAVE_CONFIG_H")
```
Comment (waruqi) [inline packages/l/libkmod/port/xmake.lua:12]: 我没看到有用到这个么。。https://github.com/search?q=repo%3Akmod-project%2Fkmod+HAVE_CONFIG_H&type=code

File: packages/l/libkmod/port/xmake.lua (hunk 3 above)
Comment (zhangxp1998) [reply packages/l/libkmod/port/xmake.lua]: done

File: packages/l/libkmod/port/xmake.lua (hunk 2 above)
Comment (zhangxp1998) [reply packages/l/libkmod/port/xmake.lua:4]: Done

File: packages/l/libkmod/port/xmake.lua (hunk 6 above)
Comment (zhangxp1998) [reply packages/l/libkmod/port/xmake.lua]: removed

File: packages/l/libkmod/port/elf.h (hunk 5 above)
Comment (zhangxp1998) [reply packages/l/libkmod/port/elf.h]: removed

**Pull request merged**